src/main/java/edu/harvard/h2ms/service/report/AbstractReportWorkerComplianceWarning.java

```diff
@@ -0,0 +1,106 @@
+package edu.harvard.h2ms.service.report;
+
+import edu.harvard.h2ms.domain.core.Event;
+import edu.harvard.h2ms.domain.core.Question;
+import edu.harvard.h2ms.domain.core.User;
+import edu.harvard.h2ms.exception.InvalidAnswerTypeException;
+import edu.harvard.h2ms.repository.QuestionRepository;
+import edu.harvard.h2ms.repository.UserRepository;
+import edu.harvard.h2ms.service.EventService;
+import edu.harvard.h2ms.service.utils.H2msRestUtils;
+import edu.harvard.h2ms.service.utils.ReportUtils;
+import edu.harvard.h2ms.service.utils.ReportUtils.NotificationFrequency;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.hibernate.Hibernate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Basis of compliance notification emails change REPORTINGINTERVAL to change the reporting scope
+ * Override getType() to change the name of the report type
+ */
+public abstract class AbstractReportWorkerComplianceWarning implements ReportWorker {
+
+  final Logger log = LoggerFactory.getLogger(AbstractReportWorkerComplianceWarning.class);
+
+  @Autowired private EventService eventService;
+
+  @Autowired private QuestionRepository questionRepository;
+
+  @Autowired private UserRepository userRepository;
+
+  long REPORTINGINTERVAL = NotificationFrequency.WEEKLY.seconds;
+
+  @Override
+  public String getType() {
+    return "abstractComplianceWarning";
+  }
+
+  @Override
+  public String createReport() {
+    List<Event> events = new ArrayList<>();
+
+    // for all questions:
+    Map<Question, Map<User, Double>> allComplianceResult = new HashMap<>();
+    Hibernate.initialize(questionRepository);
+    for (Question question : questionRepository.findAll()) {
```

Comment:
You should be able to add a method to the repository called `findAllByAnswerType` and only get boolean questions, not a big deal